Automattic/woocommerce-payments
# Allow customers to create an account automatically via Express Payment methods the subscription setting allows it

Fixes #7607 

#### Changes proposed in this Pull Request

Customers who aren't logged in can purchase products via express payment methods (eg Apple Pay) and an account will be created for them automatically, if the WooCommerce account settings allow. 

There was a bug in this flow when purchasing subscriptions. There is a separate setting that enables account creation when purchasing subscription products and our `is_account_creation_possible()` function wasn't taking into consideration that scenario. 

This PR fixes that. This means the default settings which allow guest checkout and automatic account creation for subscriptions result in the streamlined purchase flow without account creation being necessary.  

#### Testing instructions

1. Install Woo Subscriptions.
1. Go to **WooCommerce → Settings → Accounts & Privacy**
2. Make sure the **Guest checkout** and **Account creation** settings are set like this: 
   
<p align="center">
<img width="1121" alt="Screenshot 2023-11-02 at 1 37 36 pm" src="https://github.com/Automattic/woocommerce-payments/assets/8490476/4e63bd3c-51cc-4203-a9b1-079abf9414a0">

</br>
<sup><strong>Note:</strong> These are the default WooCommerce Guest and Account creation settings</sup>
</p>

4. If you don't have one already, create a subscription product.
6. Make sure Apple Pay and Google Pay are enabled in the WooPayments settings (**Payments → Settings**)
7. Go to the subscription product's page on the front end as a logged out user. 
8. Click the Apple Pay or Google Pay button. 
    - On `develop` you should see the Alert popup mentioned above.
    - On this branch you shouldn't see any popup. Completing the purchase should create an account for you.
9. Test the other scenarios. eg:
   - guest checkout with simple products.
   - with the subscription account creation setting disabled. 
   - with variable products.
   - with any combination of these settings.  



-------------------


**Post merge**
